src/Skia/Avalonia.Skia/TransformedGeometryImpl.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using Avalonia.Media;
+using Avalonia.Platform;
+using SkiaSharp;
+
+namespace Avalonia.Skia
+{
+    class TransformedGeometryImpl : GeometryImpl, ITransformedGeometryImpl
+    {
+        public TransformedGeometryImpl(GeometryImpl source, Matrix transform)
+        {
+            SourceGeometry = source;
+            Transform = transform;
+            EffectivePath = source.EffectivePath.Clone();
+            EffectivePath.Transform(transform.ToSKMatrix());
+        }
+
+        public override SKPath EffectivePath { get; }
+
+        public IGeometryImpl SourceGeometry { get; }
+
+        public Matrix Transform { get; }
+
+        public override Rect Bounds => SourceGeometry.Bounds.TransformToAABB(Transform);
+
+        public override bool FillContains(Point point)
+        {
+            // TODO: Not supported by SkiaSharp yet, so use expanded Rect
+            return GetRenderBounds(0).Contains(point);
+        }
+
+        public override Rect GetRenderBounds(Pen pen)
+        {
+            return GetRenderBounds(pen.Thickness);
+        }
+
+        public override IGeometryImpl Intersect(IGeometryImpl geometry)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool StrokeContains(Pen pen, Point point)
+        {
+            // TODO: Not supported by SkiaSharp yet, so use expanded Rect
+            return GetRenderBounds(0).Contains(point);
```

Comment:
I don't see a direct equivalent of StrokeContains in Skia API. The closest thing we could do is to call [getFillPath](https://github.com/google/skia/blob/8ad91a9bf896d728b905124847d74787aac698a7/include/core/SkPaint.h#L730) and then call contains there. Not sure how fast would that be in terms of performance

Reply:
Ok, we need to investigate this and see what we can do.